File: 2401-longest-nice-subarray/2401-longest-nice-subarray.java
class Solution {
    public int longestNiceSubarray(int[] ar) {
        int ans=0;
        for(int i=0;i<ar.length;i++){
            int dp[]=new int[32];
            boolean rs=true;
            for(int j=i;j<Integer.min(i+32,ar.length);j++){
                for(int k=0;k<32;k++){
                    int z = ar[j]>>k;
                    if((z&1)==1){
                        dp[k]+=1;
                    }
                }
                for(int k=0;k<32;k++){
                    if(dp[k]>1){
                        ans=Integer.max(ans,j-i);
                        rs=false;
                        break;
                    }
                }
                if(!rs)
                    break;
            }
            if(rs){
                ans=Integer.max(Integer.min(ar.length,i+32)-i,ans);
            }
        }
        return ans;
    }
}

File: 2401-longest-nice-subarray/2401-longest-nice-subarray.cpp
class Solution {
public:
    int longestNiceSubarray(vector<int>& nums) {
        vector<int>pos(31, 0);
        int ans = 0, last = 0;
        for(int i=0; i<nums.size(); i++){
            for(int j=0; j<31; j++){
                if((nums[i]>>j)&1LL) {
                    last = max(last, pos[j]);
                    pos[j] = i+1;
                }
            }
            ans = max(ans, i+1-last);
        }
        return ans;
    }
};